feat: threw error if profile-create-new-if-missing used without profile

akhilpanchal opened the pull request:
Follow up to #2064 based on @rpl's [comment](https://github.com/mozilla/web-ext/pull/2064#discussion_r518873064).

Throws error if `profile-create-new-if-missing` option used without `firefox-profile` or `chromium-profile`.

rpl commented on src/cmd/run.js:
```diff
@@ -131,6 +132,11 @@ export default async function run(
       log.info(`Profile directory not found. Creating directory ${profileDir}`);
       await fs.mkdir(profileDir);
     }
+  } else if (profileCreateIfMissing) {
+    throw new UsageError(
+      '--profile-create-if-missing should be paired ' +
+      'with --firefox-profile or --chromium-profile'
+    );
```
we could move this into the previous if block by doing something like:

```
if (profileCreateIfMissing) {
  if (!profileDir) {
    throw new UsageError("...);
  }
  // the rest of the if block from line 128 to line 133.
}
```

rpl commented on tests/unit/test-cmd/test.run.js:
```diff
@@ -406,5 +407,23 @@ describe('run', () => {
+       async () => {
+         const cmd = prepareRun();
+
+         let actualError = null;
+         try {
+           await cmd.run({ profileCreateIfMissing: true });
+         } catch (error) {
+           actualError = error;
+         }
+
+         assert.instanceOf(actualError, UsageError);
+         assert.match(
+           actualError && actualError.message,
+           /should be paired with --firefox-profile or --chromium-profile/
+         );
```
we should use `await assert.isRejected(...)` (provided by the chai-as-promised dev dependency and also used in some of the other tests, e.g. test.adb.js should be one of the test file already using this assertion functions).

rpl reviewed (changes requested):
@akhilpanchal Thanks for looking into quickly jumping on covering the followups related to #2064!!! 
Follows a couple of review comments.

akhilpanchal commented:
> @akhilpanchal Thanks for looking into quickly jumping on covering the followups related to #2064!!!
> Follows a couple of review comments.

Thanks @rpl. I addressed your feedback in the latest commit. 
I also have a [pull request open in the extension-workshop repo](https://github.com/mozilla/extension-workshop/pull/832) for documenting.
Let me know if you have any feedback for me.

Thanks again for all your guidance and comments :)

rpl commented on src/cmd/run.js:
```diff
@@ -123,7 +124,13 @@ export default async function run(
+  if (profileCreateIfMissing) {
+    if (!profileDir) {
+      throw new UsageError(
+        '--profile-create-if-missing should be paired ' +
```
would you mind to replace `should be paired with` with `requires`? (mostly because I'm not really sure if paired is a common why to refer to that in english, "--profile-create-if-missing requires --firefox-profile or --chromium-profile" is shorter and should be clear enough).

Same change has to be applied to the assertion on the test side.

akhilpanchal replied:
> @akhilpanchal good job, this looks great. I would just tweak the UsageError message a bit (my apologies I should have noticed it sooner and pointed it out in the previous review round) before merging this on master, would you mind to do that? (the tweak is described below).

No worries at all! Makes sense to me. It is shorter and clear.
I addressed it.

rpl reviewed (commented):
@akhilpanchal good job, this looks great. I would just tweak the UsageError message a bit (my apologies I should have noticed it sooner and pointed it out in the previous review round) before merging this on master, would you mind to do that? (the tweak is described below).

rpl commented:
@akhilpanchal Thanks a lot! looks great and I'm going to merge it right now! :tada:
